libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -596,20 +596,24 @@ - (void)switchToNewUser {
 }
 
 - (void)switchToUser:(SFUserAccount *)newCurrentUser {
-    [self enumerateDelegates:^(id<SFUserAccountManagerDelegate> delegate) {
-        if ([delegate respondsToSelector:@selector(userAccountManager:willSwitchFromUser:toUser:)]) {
-            [delegate userAccountManager:self willSwitchFromUser:self.currentUser toUser:newCurrentUser];
-        }
-    }];
-
-    SFUserAccount *prevUser = self.currentUser;
-    [self setCurrentUser:newCurrentUser];
-
-    [self enumerateDelegates:^(id<SFUserAccountManagerDelegate> delegate) {
-        if ([delegate respondsToSelector:@selector(userAccountManager:didSwitchFromUser:toUser:)]) {
-            [delegate userAccountManager:self didSwitchFromUser:prevUser toUser:newCurrentUser];
-        }
-    }];
+    if (self.currentUser == newCurrentUser) {
```

Comment:
It was a conscious choice, but I'm wondering too if I should have gone the other way.  The premise is that, with 5.1 and beyond, we should no longer have the possibility of multiple `SFUserAccount` instances with the same identity.  But in the event that we do, for whatever reason, it may make more sense to "fail fast" and disallow that in the switch as well.

I'll make the change.